### DISCUSSION TITLE
.resx file in the shared project

### DISCUSSION THREAD

**Discussion by PhilJollans** [codecadwallader/codemaid#904] [General]
This question isn't really about CodeMaid, but about how the CodeMaidShared project is used.

The **CodeMaidShared** project contains a resource file **Resources.resx**, and a code file **Resources.resx.cs**. The Custom Tool **PublicResXFileCodeGenerator** is associated with the .resx file.

In other types of projects, you can open the context menu on the .resx file and select "Run Custom Tool" to run the custom tool which generates the code file. In this project, the menu option "Run Custom Tool" is not present.

If I add a resource string to the file, the custom tool does not run automatically.

Do you have some method of generating the code file Resources.resx.cs in the shared project?
Is it possible, that you have not modified the resource file since moving it into the shared project?

My understanding, up to now, was that shared projects don't really support .resx files. 

Phil

**Comment by codecadwallader** [comment]
We moved to a shared project infrastructure recently in order to support the compatibility breaking changes in VS2022.  I don't believe we have updated any of the resources yet since that switch over.  Your understanding may be correct and something I wasn't aware of yet.  I suppose the work-around would be to hand-edit the files instead of using the tooling?

**Reply by PhilJollans** [reply]
I asked a question on Stack Overflow and now posted my own detailed reply  
https://stackoverflow.com/questions/70820056/how-to-localize-net-shared-projects

and also posted feedback in the Microsoft Developer Community  
https://developercommunity.visualstudio.com/t/Support-for-localization-in-shared-proje/1662117
